src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+    const { SubIFDs } = firstImage.fileDirectory;
+    this.numLevels =
+      this.omexml.getNumberOfImages() || (SubIFDs && SubIFDs.length);
+    this.isLegacyBioFormatsPyramid = !SubIFDs;
+    this.isPyramid = !!this.numLevels;
+    const type = this.omexml.Type;
+    if (type === 'uint8') {
+      this.dtype = '<u1';
+    }
+    if (type === 'uint16') {
+      this.dtype = '<u2';
+    }
+    if (type === 'uint32') {
+      this.dtype = '<u4';
+    }
+    if (type === 'float32') {
+      this.dtype = '<f4';
+    }
+  }
+
+  /**
+   * Returns an IFD index for a given loader selection.
+   * @param {number} z Z axis selection.
+   * @param {number} time Time axis selection.
+   * @param {String} channel Channel axis selection.
+   * @returns {number} IFD index.
+   */
+  _getIFDIndex({ z = 0, channel, time = 0 }) {
+    let channelIndex;
+    if (this.channelNames.every(v => !v)) {
+      // Without names, enforce a numeric channel indexing scheme
+      console.warn(
+        'There are no channel names in the OMEXML.  Please use numeric indexing'
+      );
+      channelIndex = channel;
+    } else {
+      channelIndex = this.channelNames.indexOf(channel);
+    }
+    const { SizeZ, SizeT, SizeC, DimensionOrder } = this.omexml;
+    switch (DimensionOrder) {
+      case 'XYZCT': {
+        return time * SizeZ * SizeC + channelIndex * SizeZ + z;
+      }
+      case 'XYZTC': {
+        return channelIndex * SizeZ * SizeT + time * SizeZ + z;
+      }
+      case 'XYCTZ': {
+        return z * SizeC * SizeT + time * SizeC + channelIndex;
+      }
+      case 'XYCZT': {
+        return time * SizeC * SizeZ + z * SizeC + channelIndex;
+      }
+      case 'XYTCZ': {
+        return z * SizeT * SizeC + channelIndex * SizeT + time;
+      }
+      case 'XYTZC': {
+        return channelIndex * SizeT * SizeZ + z * SizeT + time;
+      }
+      default: {
+        throw new Error('Dimension order is required for OMETIFF');
+      }
+    }
+  }
+
+  /**
+   * Handles `onTileError` within deck.gl
+   * @param {Error} err Error thrown in tile layer
+   */
+  // eslint-disable-next-line class-methods-use-this
+  onTileError(err) {
+    console.error(err);
+  }
+
+  /**
+   * Returns image tiles at tile-position (x, y) at pyramidal level z.
+   * @param {number} x positive integer
+   * @param {number} y positive integer
+   * @param {number} z positive integer (0 === highest zoom level)
+   * @param {Array} loaderSelection, Array of number Arrays specifying channel selections
+   * @returns {Object} data: TypedArray[], width: number (tileSize), height: number (tileSize)
+   */
+  async getTile({ x, y, z, loaderSelection }) {
+    if (!this._tileInBounds({ x, y, z })) {
+      return null;
+    }
+    const { tiff, offsets, isLegacyBioFormatsPyramid, omexml, tileSize } = this;
+    const { SizeZ, SizeT, SizeC } = omexml;
+    const pyramidOffset = z * SizeZ * SizeT * SizeC;
+    let image;
+    const tileRequests = loaderSelection.map(async index => {
```

Comment:
what is `loaderSelection`? Seems like an array of numbers?

Reply:
Yup.